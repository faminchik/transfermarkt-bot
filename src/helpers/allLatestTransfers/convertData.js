import _ from 'lodash';
import cheerio from 'cheerio';

const transposeArrays = arrays =>
    _.reduce(
        arrays,
        (acc, array) => {
            _.map(array, (item, i) => {
                if (!_.isArray(acc[i])) acc[i] = [];
                acc[i].push(item);
            });
            return acc;
        },
        []
    );

const formArrayWithConfig = (array, config) =>
    _.map(array, item =>
        _.reduce(
            item,
            (acc, param, index) => ({
                ...acc,
                [config[index]]: param
            }),
            {}
        )
    );

const validTextDataIndexes = [2, 4, 8, 12, 14, 15];
const configText = ['name', 'age', 'leftTeam', 'joinedTeam', 'transferDate', 'marketValue'];

const validHtmlDataIndexes = [5, 9, 13, 16];
const configHtml = ['nationality', 'leftTeamCountry', 'joinedTeamCountry', 'fee'];

const convertTextData = textData => {
    const compactedTextData = _.map(textData, item => _.compact(item));

    const neededTextData = _.filter(compactedTextData, (item, index) =>
        _.includes(validTextDataIndexes, index)
    );

    neededTextData[1].splice(0, 1);
    neededTextData[0] = _.filter(neededTextData[0], (player, index) => index % 2 === 1);

    const transformedTextData = transposeArrays(neededTextData);

    return formArrayWithConfig(transformedTextData, configText);
};

const convertHtmlData = htmlData => {
    const compactedHtmlData = _.map(htmlData, item => _.compact(item));

    const neededHtmlData = _.filter(compactedHtmlData, (item, index) =>
        _.includes(validHtmlDataIndexes, index)
    );

    // nationality
    neededHtmlData[0].splice(0, 1);
    neededHtmlData[0] = _.map(neededHtmlData[0], item => {
        const $ = cheerio.load(item);
        return $('img').attr('alt');
    });

    // leftTeamCountry
    neededHtmlData[1] = _.map(neededHtmlData[1], item => {
        const $ = cheerio.load(item);
        const result = $('img').attr('alt');
        return result ? result : '';
    });

    // joinedTeamCountry
    neededHtmlData[2] = _.map(neededHtmlData[2], item => {
        const $ = cheerio.load(item);
        const result = $('img').attr('alt');
        return result ? result : '';
    });

    // fee
    neededHtmlData[3] = _.map(neededHtmlData[3], item => {
        const $ = cheerio.load(item);
        return $('a').text();
    });

    const transformedHtmlData = transposeArrays(neededHtmlData);

    return formArrayWithConfig(transformedHtmlData, configHtml);
};

export default ({ textData, htmlData }) => {
    const convertedTextData = convertTextData(textData);
    const convertedHtmlData = convertHtmlData(htmlData);

    const resultData = [];
    for (let i = 0; i < _.size(convertedTextData); i++) {
        resultData.push(convertedTextData[i]);
        if (convertedHtmlData[i]) {
            resultData[i] = {
                ...resultData[i],
                ...convertedHtmlData[i]
            };
        }
    }

    return resultData;
};
