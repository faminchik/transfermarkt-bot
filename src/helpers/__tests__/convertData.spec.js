import getTableDataFromHTML from 'helpers/allLatestTransfers/getTableDataFromHTML';
import convertData from 'helpers/convertData';
import config from 'configs/convertDataConfig';
import { ALL_LATEST_TRANSFERS } from 'constants/transfermarkt/ConvertDataTypes';
import html from './data/convertData_data';

const expectedResult = [
    {
        name: 'Quentin Westberg',
        age: '32',
        leftTeam: 'AJ Auxerre',
        joinedTeam: 'Toronto FC',
        transferDate: 'Feb 25, 2019',
        marketValue: '600 Th. €',
        fee: '?',
        nationality: 'United States',
        leftTeamCountry: 'France',
        joinedTeamCountry: 'Canada'
    },
    {
        name: 'Kévin Boli',
        age: '27',
        leftTeam: 'GZ Hengfeng',
        joinedTeam: 'CFR Cluj',
        transferDate: 'Feb 24, 2019',
        marketValue: '1,20 Mill. €',
        fee: 'Loan',
        nationality: "Cote d'Ivoire",
        leftTeamCountry: 'China',
        joinedTeamCountry: 'Romania'
    },
    {
        name: 'Yuning Zhang',
        age: '22',
        leftTeam: 'West Brom',
        joinedTeam: 'BJ Sinobo Guoan',
        transferDate: 'Feb 24, 2019',
        marketValue: '1,50 Mill. €',
        fee: '?',
        nationality: 'China',
        leftTeamCountry: 'England',
        joinedTeamCountry: 'China'
    },
    {
        name: 'Anton Maglica',
        age: '27',
        leftTeam: 'Apol. Limassol',
        joinedTeam: 'GZ Hengfeng',
        transferDate: 'Feb 24, 2019',
        marketValue: '1,50 Mill. €',
        fee: '450 Th. €',
        nationality: 'Croatia',
        leftTeamCountry: 'Cyprus',
        joinedTeamCountry: 'China'
    },
    {
        name: 'Sone Aluko',
        age: '30',
        leftTeam: 'Reading',
        joinedTeam: 'BJ Renhe',
        transferDate: 'Feb 23, 2019',
        marketValue: '3,50 Mill. €',
        fee: '?',
        nationality: 'Nigeria',
        leftTeamCountry: 'England',
        joinedTeamCountry: 'China'
    },
    {
        name: 'C.J. Sapong',
        age: '30',
        leftTeam: 'Philadelphia',
        joinedTeam: 'Chicago Fire',
        transferDate: 'Feb 23, 2019',
        marketValue: '1,25 Mill. €',
        fee: '400 Th. €',
        nationality: 'United States',
        leftTeamCountry: 'United States',
        joinedTeamCountry: 'United States'
    },
    {
        name: 'Lacina Traoré',
        age: '28',
        leftTeam: 'Without Club',
        joinedTeam: 'Újpest FC',
        transferDate: 'Feb 23, 2019',
        marketValue: '2,50 Mill. €',
        fee: '-',
        nationality: "Cote d'Ivoire",
        leftTeamCountry: '',
        joinedTeamCountry: 'Hungary'
    },
    {
        name: 'Andrii Tkachuk',
        age: '31',
        leftTeam: 'Akzhayik',
        joinedTeam: 'Atyrau',
        transferDate: 'Feb 23, 2019',
        marketValue: '600 Th. €',
        fee: 'Free Transfer',
        nationality: 'Ukraine',
        leftTeamCountry: 'Kazakhstan',
        joinedTeamCountry: 'Kazakhstan'
    },
    {
        name: 'André',
        age: '28',
        leftTeam: 'Sport Recife',
        joinedTeam: 'Grêmio',
        transferDate: 'Feb 22, 2019',
        marketValue: '2,75 Mill. €',
        fee: '2,50 Mill. €',
        nationality: 'Brazil',
        leftTeamCountry: 'Brazil',
        joinedTeamCountry: 'Brazil'
    },
    {
        name: 'Alan Chochiev',
        age: '27',
        leftTeam: 'KS Samara',
        joinedTeam: 'Without Club',
        transferDate: 'Feb 22, 2019',
        marketValue: '700 Th. €',
        fee: '-',
        nationality: 'Russia',
        leftTeamCountry: 'Russia',
        joinedTeamCountry: ''
    },
    {
        name: 'Viacheslav Tankovskyi',
        age: '23',
        leftTeam: 'Shakhtar D.',
        joinedTeam: 'Arsenal Kyiv',
        transferDate: 'Feb 22, 2019',
        marketValue: '500 Th. €',
        fee: 'Loan',
        nationality: 'Ukraine',
        leftTeamCountry: 'Ukraine',
        joinedTeamCountry: 'Ukraine'
    },
    {
        name: 'Javokhir Sidikov',
        age: '22',
        leftTeam: 'Kokand 1912',
        joinedTeam: 'Pakhtakor',
        transferDate: 'Feb 22, 2019',
        marketValue: '600 Th. €',
        fee: '?',
        nationality: 'Uzbekistan',
        leftTeamCountry: 'Uzbekistan',
        joinedTeamCountry: 'Uzbekistan'
    },
    {
        name: 'José León',
        age: '24',
        leftTeam: 'Rayo Vallecano',
        joinedTeam: 'AFC Eskilstuna',
        transferDate: 'Feb 22, 2019',
        marketValue: '800 Th. €',
        fee: 'Loan',
        nationality: 'Spain',
        leftTeamCountry: 'Spain',
        joinedTeamCountry: 'Sweden'
    },
    {
        name: 'Muris Mesanovic',
        age: '28',
        leftTeam: 'Slavia Prag',
        joinedTeam: 'Mladá Boleslav',
        transferDate: 'Feb 22, 2019',
        marketValue: '600 Th. €',
        fee: '?',
        nationality: 'Bosnia-Herzegovina',
        leftTeamCountry: 'Czech Republic',
        joinedTeamCountry: 'Czech Republic'
    },
    {
        name: 'Jan Sykora',
        age: '25',
        leftTeam: 'Slavia Prag',
        joinedTeam: 'Slovan Liberec',
        transferDate: 'Feb 22, 2019',
        marketValue: '1,25 Mill. €',
        fee: 'Loan',
        nationality: 'Czech Republic',
        leftTeamCountry: 'Czech Republic',
        joinedTeamCountry: 'Czech Republic'
    },
    {
        name: 'Jakub Jugas',
        age: '26',
        leftTeam: 'Slavia Prag',
        joinedTeam: 'Mladá Boleslav',
        transferDate: 'Feb 22, 2019',
        marketValue: '1,25 Mill. €',
        fee: 'Loan',
        nationality: 'Czech Republic',
        leftTeamCountry: 'Czech Republic',
        joinedTeamCountry: 'Czech Republic'
    },
    {
        name: 'Filip Panak',
        age: '23',
        leftTeam: 'MFK Karvina',
        joinedTeam: 'Sparta Praha',
        transferDate: 'Feb 22, 2019',
        marketValue: '800 Th. €',
        fee: '?',
        nationality: 'Czech Republic',
        leftTeamCountry: 'Czech Republic',
        joinedTeamCountry: 'Czech Republic'
    },
    {
        name: 'Timofey Margasov',
        age: '26',
        leftTeam: 'Loko Moscow',
        joinedTeam: 'Sochi',
        transferDate: 'Feb 22, 2019',
        marketValue: '500 Th. €',
        fee: 'Loan',
        nationality: 'Russia',
        leftTeamCountry: 'Russia',
        joinedTeamCountry: 'Russia'
    },
    {
        name: 'Dorin Rotariu',
        age: '23',
        leftTeam: 'Club Brugge',
        joinedTeam: 'FC Astana',
        transferDate: 'Feb 22, 2019',
        marketValue: '1,00 Mill. €',
        fee: '?',
        nationality: 'Romania',
        leftTeamCountry: 'Belgium',
        joinedTeamCountry: 'Kazakhstan'
    },
    {
        name: 'David Stockdale',
        age: '33',
        leftTeam: 'Coventry',
        joinedTeam: 'Birmingham',
        transferDate: 'Feb 22, 2019',
        marketValue: '1,00 Mill. €',
        fee: 'End of loan',
        nationality: 'England',
        leftTeamCountry: 'England',
        joinedTeamCountry: 'England'
    },
    {
        name: 'Borek Dockal',
        age: '30',
        leftTeam: 'HN Jianye',
        joinedTeam: 'Sparta Praha',
        transferDate: 'Feb 22, 2019',
        marketValue: '2,00 Mill. €',
        fee: '1,30 Mill. €',
        nationality: 'Czech Republic',
        leftTeamCountry: 'China',
        joinedTeamCountry: 'Czech Republic'
    },
    {
        name: 'Mohammed Rabiu',
        age: '29',
        leftTeam: 'Anzhi',
        joinedTeam: 'KS Samara',
        transferDate: 'Feb 22, 2019',
        marketValue: '600 Th. €',
        fee: 'Free Transfer',
        nationality: 'Ghana',
        leftTeamCountry: 'Russia',
        joinedTeamCountry: 'Russia'
    },
    {
        name: 'Yuning Zhang',
        age: '22',
        leftTeam: 'ADO Den Haag',
        joinedTeam: 'West Brom',
        transferDate: 'Feb 22, 2019',
        marketValue: '1,50 Mill. €',
        fee: 'End of loan',
        nationality: 'China',
        leftTeamCountry: 'Netherlands',
        joinedTeamCountry: 'England'
    },
    {
        name: 'Filipe Nascimento',
        age: '24',
        leftTeam: 'Levski Sofia',
        joinedTeam: 'Poli. Iasi',
        transferDate: 'Feb 21, 2019',
        marketValue: '700 Th. €',
        fee: 'Loan',
        nationality: 'Portugal',
        leftTeamCountry: 'Bulgaria',
        joinedTeamCountry: 'Romania'
    },
    {
        name: 'Dylan Flores',
        age: '25',
        leftTeam: 'FC Poli Iasi',
        joinedTeam: 'Sepsi Sf. Gh.',
        transferDate: 'Feb 21, 2019',
        marketValue: '700 Th. €',
        fee: '90 Th. €',
        nationality: 'Costa Rica',
        leftTeamCountry: 'Romania',
        joinedTeamCountry: 'Romania'
    }
];

describe('convertData', () => {
    describe('allLatestTransfers', () => {
        test('should convert html data in a correct way', () => {
            const { textData, htmlData } = getTableDataFromHTML(html);
            const transfersInfo = convertData({ textData, htmlData }, config[ALL_LATEST_TRANSFERS]);

            expect(transfersInfo).toEqual(expectedResult);
        });
    });
});
