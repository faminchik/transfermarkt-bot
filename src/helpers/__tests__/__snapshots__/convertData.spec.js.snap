// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertData allLatestTransfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "32",
    "fee": "?",
    "joinedTeam": "Toronto FC",
    "joinedTeamCountry": "Canada",
    "leftTeam": "AJ Auxerre",
    "leftTeamCountry": "France",
    "marketValue": "600 Th. €",
    "name": "Quentin Westberg",
    "nationality": "United States",
    "transferDate": "Feb 25, 2019",
  },
  Object {
    "age": "27",
    "fee": "Loan",
    "joinedTeam": "CFR Cluj",
    "joinedTeamCountry": "Romania",
    "leftTeam": "GZ Hengfeng",
    "leftTeamCountry": "China",
    "marketValue": "1,20 Mill. €",
    "name": "Kévin Boli",
    "nationality": "Cote d'Ivoire",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "22",
    "fee": "?",
    "joinedTeam": "BJ Sinobo Guoan",
    "joinedTeamCountry": "China",
    "leftTeam": "West Brom",
    "leftTeamCountry": "England",
    "marketValue": "1,50 Mill. €",
    "name": "Yuning Zhang",
    "nationality": "China",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "27",
    "fee": "450 Th. €",
    "joinedTeam": "GZ Hengfeng",
    "joinedTeamCountry": "China",
    "leftTeam": "Apol. Limassol",
    "leftTeamCountry": "Cyprus",
    "marketValue": "1,50 Mill. €",
    "name": "Anton Maglica",
    "nationality": "Croatia",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "30",
    "fee": "?",
    "joinedTeam": "BJ Renhe",
    "joinedTeamCountry": "China",
    "leftTeam": "Reading",
    "leftTeamCountry": "England",
    "marketValue": "3,50 Mill. €",
    "name": "Sone Aluko",
    "nationality": "Nigeria",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "30",
    "fee": "",
    "joinedTeam": "Chicago Fire",
    "joinedTeamCountry": "United States",
    "leftTeam": "Philadelphia",
    "leftTeamCountry": "United States",
    "marketValue": "1,25 Mill. €",
    "name": "C.J. Sapong",
    "nationality": "United States",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "28",
    "fee": "-",
    "joinedTeam": "Újpest FC",
    "joinedTeamCountry": "Hungary",
    "leftTeam": "Without Club",
    "leftTeamCountry": "",
    "marketValue": "2,50 Mill. €",
    "name": "Lacina Traoré",
    "nationality": "Cote d'Ivoire",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "31",
    "fee": "Free Transfer",
    "joinedTeam": "Atyrau",
    "joinedTeamCountry": "Kazakhstan",
    "leftTeam": "Akzhayik",
    "leftTeamCountry": "Kazakhstan",
    "marketValue": "600 Th. €",
    "name": "Andrii Tkachuk",
    "nationality": "Ukraine",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "28",
    "fee": "2,50 Mill. €",
    "joinedTeam": "Grêmio",
    "joinedTeamCountry": "Brazil",
    "leftTeam": "Sport Recife",
    "leftTeamCountry": "Brazil",
    "marketValue": "2,75 Mill. €",
    "name": "André",
    "nationality": "Brazil",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "27",
    "fee": "-",
    "joinedTeam": "Without Club",
    "joinedTeamCountry": "",
    "leftTeam": "KS Samara",
    "leftTeamCountry": "Russia",
    "marketValue": "700 Th. €",
    "name": "Alan Chochiev",
    "nationality": "Russia",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "Loan",
    "joinedTeam": "Arsenal Kyiv",
    "joinedTeamCountry": "Ukraine",
    "leftTeam": "Shakhtar D.",
    "leftTeamCountry": "Ukraine",
    "marketValue": "500 Th. €",
    "name": "Viacheslav Tankovskyi",
    "nationality": "Ukraine",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "22",
    "fee": "?",
    "joinedTeam": "Pakhtakor",
    "joinedTeamCountry": "Uzbekistan",
    "leftTeam": "Kokand 1912",
    "leftTeamCountry": "Uzbekistan",
    "marketValue": "600 Th. €",
    "name": "Javokhir Sidikov",
    "nationality": "Uzbekistan",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "24",
    "fee": "Loan",
    "joinedTeam": "AFC Eskilstuna",
    "joinedTeamCountry": "Sweden",
    "leftTeam": "Rayo Vallecano",
    "leftTeamCountry": "Spain",
    "marketValue": "800 Th. €",
    "name": "José León",
    "nationality": "Spain",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "28",
    "fee": "?",
    "joinedTeam": "Mladá Boleslav",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "600 Th. €",
    "name": "Muris Mesanovic",
    "nationality": "Bosnia-Herzegovina",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "25",
    "fee": "Loan",
    "joinedTeam": "Slovan Liberec",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "1,25 Mill. €",
    "name": "Jan Sykora",
    "nationality": "Czech Republic",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "26",
    "fee": "Loan",
    "joinedTeam": "Mladá Boleslav",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "1,25 Mill. €",
    "name": "Jakub Jugas",
    "nationality": "Czech Republic",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "?",
    "joinedTeam": "Sparta Praha",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "MFK Karvina",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "800 Th. €",
    "name": "Filip Panak",
    "nationality": "Czech Republic",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "26",
    "fee": "Loan",
    "joinedTeam": "Sochi",
    "joinedTeamCountry": "Russia",
    "leftTeam": "Loko Moscow",
    "leftTeamCountry": "Russia",
    "marketValue": "500 Th. €",
    "name": "Timofey Margasov",
    "nationality": "Russia",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "?",
    "joinedTeam": "FC Astana",
    "joinedTeamCountry": "Kazakhstan",
    "leftTeam": "Club Brugge",
    "leftTeamCountry": "Belgium",
    "marketValue": "1,00 Mill. €",
    "name": "Dorin Rotariu",
    "nationality": "Romania",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "33",
    "fee": "End of loan",
    "joinedTeam": "Birmingham",
    "joinedTeamCountry": "England",
    "leftTeam": "Coventry",
    "leftTeamCountry": "England",
    "marketValue": "1,00 Mill. €",
    "name": "David Stockdale",
    "nationality": "England",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "30",
    "fee": "1,30 Mill. €",
    "joinedTeam": "Sparta Praha",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "HN Jianye",
    "leftTeamCountry": "China",
    "marketValue": "2,00 Mill. €",
    "name": "Borek Dockal",
    "nationality": "Czech Republic",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "29",
    "fee": "Free Transfer",
    "joinedTeam": "KS Samara",
    "joinedTeamCountry": "Russia",
    "leftTeam": "Anzhi",
    "leftTeamCountry": "Russia",
    "marketValue": "600 Th. €",
    "name": "Mohammed Rabiu",
    "nationality": "Ghana",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "22",
    "fee": "End of loan",
    "joinedTeam": "West Brom",
    "joinedTeamCountry": "England",
    "leftTeam": "ADO Den Haag",
    "leftTeamCountry": "Netherlands",
    "marketValue": "1,50 Mill. €",
    "name": "Yuning Zhang",
    "nationality": "China",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "24",
    "fee": "Loan",
    "joinedTeam": "Poli. Iasi",
    "joinedTeamCountry": "Romania",
    "leftTeam": "Levski Sofia",
    "leftTeamCountry": "Bulgaria",
    "marketValue": "700 Th. €",
    "name": "Filipe Nascimento",
    "nationality": "Portugal",
    "transferDate": "Feb 21, 2019",
  },
  Object {
    "age": "25",
    "fee": "90 Th. €",
    "joinedTeam": "Sepsi Sf. Gh.",
    "joinedTeamCountry": "Romania",
    "leftTeam": "FC Poli Iasi",
    "leftTeamCountry": "Romania",
    "marketValue": "700 Th. €",
    "name": "Dylan Flores",
    "nationality": "Costa Rica",
    "transferDate": "Feb 21, 2019",
  },
]
`;

exports[`convertData search clubsSearch should convert html data in a correct way 1`] = `
Array [
  Object {
    "clubLink": "/inter-mailand/startseite/verein/46",
    "clubName": "Inter Milan",
    "country": "Italy",
    "totalMarketValue": "557,40 Mill. €",
  },
  Object {
    "clubLink": "/ac-mailand/startseite/verein/5",
    "clubName": "AC Milan",
    "country": "Italy",
    "totalMarketValue": "530,00 Mill. €",
  },
  Object {
    "clubLink": "/fk-metalac-gornji-milanovac/startseite/verein/12135",
    "clubName": "FK Metalac Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "3,05 Mill. €",
  },
  Object {
    "clubLink": "/inter-mailand-primavera/startseite/verein/5380",
    "clubName": "Inter Milan Primavera",
    "country": "Italy",
    "totalMarketValue": "2,20 Mill. €",
  },
  Object {
    "clubLink": "/inter-mailand-uefa-u19/startseite/verein/62104",
    "clubName": "Inter Mailand UEFA U19",
    "country": "Italy",
    "totalMarketValue": "2,20 Mill. €",
  },
  Object {
    "clubLink": "/us-bustese-roncalli/startseite/verein/49933",
    "clubName": "Milano City FC",
    "country": "Italy",
    "totalMarketValue": "700 Th. €",
  },
  Object {
    "clubLink": "/ac-mailand-primavera/startseite/verein/10958",
    "clubName": "Milan Primavera",
    "country": "Italy",
    "totalMarketValue": "400 Th. €",
  },
  Object {
    "clubLink": "/milano-united-a-f-c-/startseite/verein/37588",
    "clubName": "Milano Unite FC",
    "country": "South Africa",
    "totalMarketValue": "250 Th. €",
  },
  Object {
    "clubLink": "/fk-takovo-gornji-milanovac/startseite/verein/32117",
    "clubName": "FK Takovo Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "100 Th. €",
  },
  Object {
    "clubLink": "/fk-lunjevica-gornji-milanovac/startseite/verein/71902",
    "clubName": "FK Lunjevica Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "100 Th. €",
  },
]
`;

exports[`convertData teamTransfers arrivals transfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "19",
    "fee": "135,00 Mill. €",
    "marketValue": "120,00 Mill. €",
    "name": "Kylian Mbappé",
    "nationality": "France",
    "secondPartyTeam": "Monaco",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "21",
    "fee": "37,00 Mill. €",
    "marketValue": "18,00 Mill. €",
    "name": "Thilo Kehrer",
    "nationality": "Germany",
    "secondPartyTeam": "FC Schalke 04",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "25",
    "fee": "5,00 Mill. €",
    "marketValue": "12,00 Mill. €",
    "name": "Juan Bernat",
    "nationality": "Spain",
    "secondPartyTeam": "Bayern Munich",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "40",
    "fee": "Free Transfer",
    "marketValue": "2,00 Mill. €",
    "name": "Gianluigi Buffon",
    "nationality": "Italy",
    "secondPartyTeam": "Juventus",
    "secondPartyTeamCountry": "Italy",
  },
  Object {
    "age": "29",
    "fee": "Free Transfer",
    "marketValue": "8,00 Mill. €",
    "name": "Eric Maxim Choupo-Moting",
    "nationality": "Cameroon",
    "secondPartyTeam": "Stoke City",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Colin Dagba",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Antoine Bernede",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Stanley N'Soki",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "18",
    "fee": "-",
    "marketValue": "500 Th. €",
    "name": "Moussa Diaby",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "18",
    "fee": "-",
    "marketValue": "1,00 Mill. €",
    "name": "Timothy Weah",
    "nationality": "United States",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "17",
    "fee": "-",
    "marketValue": "-",
    "name": "Yacine Adli",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "20",
    "fee": "End of loan",
    "marketValue": "700 Th. €",
    "name": "Jonathan Ikoné",
    "nationality": "France",
    "secondPartyTeam": "Montpellier",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "28",
    "fee": "End of loan",
    "marketValue": "18,00 Mill. €",
    "name": "Grzegorz Krychowiak",
    "nationality": "Poland",
    "secondPartyTeam": "West Brom",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "22",
    "fee": "End of loan",
    "marketValue": "600 Th. €",
    "name": "Rémy Descamps",
    "nationality": "France",
    "secondPartyTeam": "Tours FC",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "25",
    "fee": "End of loan",
    "marketValue": "7,50 Mill. €",
    "name": "Jesé",
    "nationality": "Spain",
    "secondPartyTeam": "Stoke City",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "21",
    "fee": "End of loan",
    "marketValue": "40,00 Mill. €",
    "name": "Gonçalo Guedes",
    "nationality": "Portugal",
    "secondPartyTeam": "Valencia",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "25",
    "fee": "End of loan",
    "marketValue": "1,50 Mill. €",
    "name": "Jean-Christophe Bahebeck",
    "nationality": "France",
    "secondPartyTeam": "FC Utrecht",
    "secondPartyTeamCountry": "Netherlands",
  },
  Object {
    "age": "20",
    "fee": "End of loan",
    "marketValue": "1,50 Mill. €",
    "name": "Odsonne Edouard",
    "nationality": "France",
    "secondPartyTeam": "Celtic",
    "secondPartyTeamCountry": "Scotland",
  },
]
`;

exports[`convertData teamTransfers departures transfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "21",
    "fee": "40,00 Mill. €",
    "marketValue": "40,00 Mill. €",
    "name": "Gonçalo Guedes",
    "nationality": "Portugal",
    "secondPartyTeam": "Valencia",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "29",
    "fee": "24,70 Mill. €",
    "marketValue": "15,00 Mill. €",
    "name": "Javier Pastore",
    "nationality": "Argentina",
    "secondPartyTeam": "AS Roma",
    "secondPartyTeamCountry": "Italy",
  },
  Object {
    "age": "28",
    "fee": "24,00 Mill. €",
    "marketValue": "17,00 Mill. €",
    "name": "Yuri Berchiche",
    "nationality": "Spain",
    "secondPartyTeam": "Athletic",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "20",
    "fee": "10,30 Mill. €",
    "marketValue": "1,50 Mill. €",
    "name": "Odsonne Edouard",
    "nationality": "France",
    "secondPartyTeam": "Celtic",
    "secondPartyTeamCountry": "Scotland",
  },
  Object {
    "age": "20",
    "fee": "5,00 Mill. €",
    "marketValue": "700 Th. €",
    "name": "Jonathan Ikoné",
    "nationality": "France",
    "secondPartyTeam": "LOSC Lille",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "22",
    "fee": "Loan fee:",
    "marketValue": "30,00 Mill. €",
    "name": "Giovani Lo Celso",
    "nationality": "Argentina",
    "secondPartyTeam": "Real Betis",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "28",
    "fee": "Loan fee:",
    "marketValue": "18,00 Mill. €",
    "name": "Grzegorz Krychowiak",
    "nationality": "Poland",
    "secondPartyTeam": "Loko Moscow",
    "secondPartyTeamCountry": "Russia",
  },
  Object {
    "age": "25",
    "fee": "Free Transfer",
    "marketValue": "1,50 Mill. €",
    "name": "Jean-Christophe Bahebeck",
    "nationality": "France",
    "secondPartyTeam": "FC Utrecht",
    "secondPartyTeamCountry": "Netherlands",
  },
  Object {
    "age": "28",
    "fee": "Loan",
    "marketValue": "8,00 Mill. €",
    "name": "Kevin Trapp",
    "nationality": "Germany",
    "secondPartyTeam": "E. Frankfurt",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "22",
    "fee": "Loan",
    "marketValue": "600 Th. €",
    "name": "Rémy Descamps",
    "nationality": "France",
    "secondPartyTeam": "Clermont Foot",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "31",
    "fee": "-",
    "marketValue": "5,00 Mill. €",
    "name": "Hatem Ben Arfa",
    "nationality": "France",
    "secondPartyTeam": "Without Club",
    "secondPartyTeamCountry": "",
  },
  Object {
    "age": "35",
    "fee": "-",
    "marketValue": "1,00 Mill. €",
    "name": "Thiago Motta",
    "nationality": "Italy",
    "secondPartyTeam": "Retired",
    "secondPartyTeamCountry": "",
  },
  Object {
    "age": "19",
    "fee": "End of loan",
    "marketValue": "120,00 Mill. €",
    "name": "Kylian Mbappé",
    "nationality": "France",
    "secondPartyTeam": "Monaco",
    "secondPartyTeamCountry": "France",
  },
]
`;
